uniconnect-epfl/uniconnect
# feat: user qr scanning finalization

# What I did
- Redirected to the `ExternalProfileScreen` when scanning someone's QR code.
- Made it so `ExternalProfileScreen` adapts if we are friend or not with the external profile (right now hard coded but will adapt to the database later). If we are not friends the screen will have a button "Add", if we are already friends we will have a button "Message" and "Remove"

# How I did it
- Deleted the `AddContactScreen` file and tests.
- Redirected the navigation and the deep linking to the `ExternalProfileScreen`.
- Modified the `ExternalProfileScreen` to fetch the user data and the external profile from the database and adapt to if the user and the external are friends or not. The adding and removal of friends will be implemented when the database can support if.

# How to verify it
You can scroll the contact list and click on contacts to see details. You can also try to scan a QR code generated with your expo go session (from inside or outside the app). You can of course also run the tests. If you want to test when you aren't friend with someone you can just modify the line 65 of `ExternalProfileScreen` and set isFriend to false.

I looked a bit into why from a version to another the QR weren't working and here is the reason. This means you need the QR to be generated into the same expo go session you are scanning from. We of course won't have this problem in production build.

<img width="500" alt="Capture" src="https://github.com/uniconnect-epfl/uniconnect/assets/93329823/62d61a63-05f1-4a6e-a5de-5bb70e31c49f">

You can read more at [this link](https://docs.expo.dev/guides/linking/)

# Demo video

https://github.com/uniconnect-epfl/uniconnect/assets/93329823/8c8bee91-8b89-4e2b-9e84-6692ac059324

# Pre-merge checklist
The changes I have introduced:
